compiler/packages/babel-plugin-react-compiler/src/HIR/Globals.ts

```diff
@@ -101,6 +90,23 @@ const TYPED_GLOBALS: Array<[string, BuiltInType]> = [
           returnValueKind: ValueKind.Mutable,
         }),
       ],
+      [
```

Comment:
Added this definition for unit test (see newly added `repro-array-map-known-mutate-shape.tsx`). It was hard finding a global function that (1) has no mutable receiver, (2) only takes one argument, and (3) mutates its argument.